dialect_cockroach.go

```diff
@@ -96,6 +96,15 @@ func (p *cockroach) Create(c *Connection, model *Model, cols columns.Columns) er
 		if err := rows.Scan(&id); err != nil {
 			return fmt.Errorf("named insert: scan: %w", err)
 		}
+		if rows.Next() {
```

Comment:
I tried to create a test for this but couldn't reproduce this locally. But the error checking definitely needs to happen, see https://pkg.go.dev/database/sql#Rows.Err. We've encountered this problem during testing. We inserted a row in CockroachDB which would violate a foreign key constraint. `rows.Scan` returned the generated UUID without erroring, but the row was not actually inserted. Checking `rows.Err()` explicitly catches this error.

Reply:
What I was wondering about this line is not about `Err()` but about `Next()`. When the DBMS returns more than one raw for this `INSERT`?

Reply:
adding `defer` for rows closing is not a blocker, adding redundant closing and/or error checking could be acceptable, but the reason for this `rows.Next()` is a critical blocker for this PR because we may forget about this discussion, or some other (including me) don't know why this block is here, then they will remove this part soon when they work on here.

If this code can be accepted, we should have a reason for this. e.g. if this code is required for now because a version of the database engine or client has a bug, we need the link for the issue here.